src/main/java/co/cask/spanner/source/SpannerInputFormat.java

```diff
@@ -0,0 +1,90 @@
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+
+/**
+ * Spanner input format, deserialize's and gets the partitions list from configuration to create input splits.
+ */
+public class SpannerInputFormat extends InputFormat<NullWritable, ResultSet> {
+  private static final Logger LOG = LoggerFactory.getLogger(SpannerInputFormat.class);
+
+  /**
+   * Deserialize batch transaction id and partitions list from configuration
+   * and create {@link PartitionInputSplit} using them.
+   */
+  @Override
+  public List<InputSplit> getSplits(JobContext jobContext) throws IOException, InterruptedException {
+    Configuration configuration = jobContext.getConfiguration();
+    List<InputSplit> partitionSplits = new ArrayList<>();
+    try {
+      BatchTransactionId batchTransactionId =
+        (BatchTransactionId) deserializeObject(configuration, SpannerConstants.SPANNER_BATCH_TRANSACTION_ID);
+      ArrayList<Partition> partitions =
+        (ArrayList<Partition>) deserializeObject(configuration, SpannerConstants.PARTITIONS_LIST);
+      for (Partition partitionrt : partitions) {
+        PartitionInputSplit split = new PartitionInputSplit(jobContext.getConfiguration(), partitionrt,
+                                                            batchTransactionId);
+        partitionSplits.add(split);
+      }
+    } catch (Exception e) {
+      throw new IOException("Exception while trying to initialize spanner and create partition splits", e);
+    }
+    LOG.debug("Initialized and configured {} splits", partitionSplits.size());
+    return partitionSplits;
+  }
+
+  private Object deserializeObject(Configuration configuration, String property) throws IOException {
+    String batchTransactionIdString = configuration.get(property);
+    if (batchTransactionIdString == null) {
+      throw new IOException(String.format("Spanner %s is not available in the configuration", property));
+    }
+    byte[] transactionIdBytes = Base64.getDecoder().decode(batchTransactionIdString);
+    try (ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(transactionIdBytes);
+         ObjectInputStream objectInputStream = new ObjectInputStream(byteArrayInputStream)) {
+      return objectInputStream.readObject();
+    } catch (ClassNotFoundException cfe) {
+      throw new IOException("Exception while trying to deserialize object ", cfe);
+    }
+  }
+
+  @Override
+  public RecordReader<NullWritable, ResultSet> createRecordReader(
+    InputSplit inputSplit, TaskAttemptContext taskAttemptContext) throws IOException, InterruptedException {
+    return new SpannerRecordReader();
```

Comment:
Here you should be getting the transaction id from the configuration and passing it to the record reader.